### ISSUE TITLE
Add option to warn on local variables that shadow allowed globals

### ISSUE THREAD

**Issue by arichard4** [mpeterv/luacheck#121]
The intent is to catch code like this: 

```
allowed_global = "whatever"

local function func(allowed_global)
	 print(allowed_global)
end

func(3)

local t = {1,2,3,4}
for allowed_global,_ in pairs(t) do
    print(allowed_global)
end
```

```
[arichard@arichard Documents]$ luacheck text.lua --new-globals="allowed_global"
Checking text.lua                                 OK

Total: 0 warnings / 0 errors in 1 file
```

A few ways that this could be done:
* Allow config to submit a list of banned local names
* Have a flag to ban local usages of permitted globals (perhaps separate flags for read-only/read-write?)

**Comment by mpeterv** [comment]
This was also suggested in #39. The main problem is that stuff like `local allowed_global = allowed_global` should be valid. And even when the value of the shadowing local is not exactly the corresponding global, the code could be perfectly valid, like `local unpack = table.unpack or unpack` or this:

```lua
local old_xpcall = xpcall
local function xpcall(func, handler, ...) -- Shadows global xpcall with version that behaves like xpcall in Lua 5,2+ even under Lua 5.1
   local args = {...}
   local num_args = select("#", ...)
   return old_xpcall(function() return func(unpack(args, 1, num_args)) end, handler)
end
```

So, in general, when the purpose of shadowing is establishing a local that does same thing as the global, perhaps with some extra features or compatibility, it probably should not be a warning. But figuring this out in the general case is difficult.

**Comment by mpeterv** [comment]
Nothing changed here. Figuring out whether the shadowing local is compatible with the global in some way requires more sophisticated analysis that what luacheck does right now.
